src/libopensc/card-iasecc.c

```diff
@@ -1270,6 +1270,9 @@ iasecc_process_fci(struct sc_card *card, struct sc_file *file,
 		}
 
 		offs++;
+		/* avoid any access to acls[offs] beyond the taglen */
+		if (offs >= (int)taglen)
+			break;
```

Comment:
I would probably add this condition to the beginning of the cycle (to avoid the issue mentioned above) and add some log message which would simplify debugging if things would go wrong.

Reply:
exactly! This check belongs into line 1247, before anything from acls is accessed. here, offs is 1 in the very first run and then increased.

Reply:
Done: the warning is only for the case of length 0, which would be strange because the mask should always be the first byte.